tfhe/src/integer/server_key/radix_parallel/tests_cases_comparisons.rs

```diff
@@ -848,75 +777,88 @@ create_parametrized_test!(integer_signed_min_parallelized_128_bits {
 ///
 /// This calls the `unchecked_scalar_server_key_method` with fresh ciphertexts
 /// and compares that it gives the same results as the `clear_fn`.
-fn test_unchecked_scalar_function<UncheckedFn, ClearF>(
+fn test_unchecked_scalar_function<UncheckedFn, ClearF, T>(
     param: ClassicPBSParameters,
     num_test: usize,
     unchecked_comparison_method: UncheckedFn,
     clear_fn: ClearF,
 ) where
-    UncheckedFn: for<'a, 'b> Fn(&'a ServerKey, &'a RadixCiphertext, U256) -> RadixCiphertext,
-    ClearF: Fn(U256, U256) -> U256,
+    T: UnsignedNumeric + DecomposableInto<u64> + RecomposableFrom<u64>,
+    UncheckedFn: for<'a, 'b> Fn(&'a ServerKey, &'a RadixCiphertext, T) -> RadixCiphertext,
+    ClearF: Fn(T, T) -> T,
+    Standard: Distribution<T>,
 {
     let mut rng = rand::thread_rng();
 
-    let num_block = (256f64 / (param.message_modulus.0 as f64).log(2.0)).ceil() as usize;
-
     let (cks, sks) = KEY_CACHE.get_from_params(param, IntegerKeyKind::Radix);
+    let num_block = divide_ceil(T::BITS, param.message_modulus.0.ilog2() as usize);
+    assert_eq!(
+        T::BITS as u32 % param.message_modulus.0.ilog2(),
+        0,
+        "bit width must be a multiple of number of bit in a block"
+    );
```

Comment:
This  assert can be removed from unchecked/unchecked_scalar tests, but should be kept in default and smart tests

Reply:
hmm, why ? don't some default or smart ops call unchecked underneath ?

Reply:
It's still the overflow thing, with 3_3 we cannot represent a u8 or u128, or whatever that is a 2^n bits thing we always have more bits.

So in default and smart tests, since we test on non-clean ciphertexts to check that the calls to full_propagate is correctly made, since we actually have more bits the clear 2^n value may overflow while the actual ciphertext won't, so we won't be comparing the same numbers between the cleans and the cipheretxts.

So that assert is there to ensure we don't call these test with block that does not have a number of bits that is a multiple of the total wanted bitsize

Reply:
yes I get that but unchecked could be called on non clean ciphertext ? or does unchecked require clean carries ?

Reply:
it requires clean ciphertext lile almost every non arithmethic operation